sources/integration/kms/kms_test.go

```diff
@@ -153,4 +156,60 @@ func KMS(t *testing.T) {
 	if searchAliasName != genAliasName() {
 		t.Fatalf("expected alias %v, got %v", genAliasName(), searchAliasName)
 	}
+
+	// List grants is not supported
+	sdpListGrants, err := grantSource.List(context.Background(), scope, true)
+	if err == nil {
+		t.Fatal("expected error but got nil")
+	}
+
+	if len(sdpListGrants) != 0 {
+		t.Fatalf("expected 0 grants, got %v", len(sdpListGrants))
+	}
+
+	// Search grants
+	sdpSearchGrants, err := grantSource.Search(context.Background(), scope, keyID, true)
+	if err != nil {
+		if strings.Contains(err.Error(), "unsupported principal type") {
```

Comment:
My principal is an assumed role type.
Because, currently there is no source to support that type, this Search call fails.
But, it may succeed if the principal is iam-role or iam-user type.